### DISCUSSION TITLE
Add support for .tsx file extension (TypeScript-JSX)

### DISCUSSION THREAD

**Discussion by danjguzman** [phcode-dev/phoenix#975] [Q&A] [answered]
When creating a new **React App + TypeScript**, the extensions of the files by default are **.tsx**.

When opening .tsx files in Phoenix, it defaults to **Text** which removes all highlighting. And sometimes (in my experience) it prevents the ability to change the file type to something else unless I close the app and try again.

![Screen Shot 2023-02-15 at 2 58 06 PM](https://user-images.githubusercontent.com/33531668/219208019-4734d158-0a15-45cb-92eb-e23bbe6ae79c.png)

**TSX** = Typescript-JSX which is not exactly the same as individual **JSX** or **TypeScript** types.

When switching to the nearest file type/syntax, **Typescript**, the highlighting is broken.

![Screen Shot 2023-02-15 at 2 44 37 PM](https://user-images.githubusercontent.com/33531668/219205834-2ae5a64e-80ca-4096-ad72-b0771bfefdcd.png)

Switching to JSX seems like it might help, but it's the wrong type for .TSX.

Brackets had a 3rd party extension for TypeScript-JSX (TSX) type which is long outdated, but might help provide some context.

![Screen Shot 2023-02-15 at 2 52 58 PM](https://user-images.githubusercontent.com/33531668/219207116-247f63f4-96c8-49e2-b2b4-a3dba2032fcf.png)

![Screen Shot 2023-02-15 at 2 56 36 PM](https://user-images.githubusercontent.com/33531668/219207754-639ae39e-0741-41ff-8637-bef769a95865.png)

**Comment by abose** [comment]
Please track in https://github.com/phcode-dev/phoenix/issues/166

**Comment by abose** [comment, marked as answer]
TS, TSX support is now available in desktop apps https://docs.phcode.dev/docs/Features/js-ts-code-intelligence
Closing as fixed
